StripeConnect/StripeConnect/Source/Internal/FinancialConnectionsPresenter.swift

```diff
@@ -9,21 +9,35 @@
 import StripeFinancialConnections
 import UIKit
 
+protocol FinancialConnectionSheetImplementation: AnyObject {
+    init(financialConnectionsSessionClientSecret: String, returnURL: String?)
+    var apiClient: STPAPIClient { get set}
+    func presentForToken(
+        from presentingViewController: UIViewController,
+        completion: @escaping (FinancialConnectionsSheet.TokenResult) -> Void
+    )
+}
```

Comment:
Instead of creating a DI protocol, you could add a separate unit-testable function that is a factory method for the sheet:
```
class FinancialConnectionsPresenter {
    func makeSheet(
        componentManager: EmbeddedComponentManager,
        clientSecret: String,
        connectedAccountId: String
    ) -> FinancialConnectionsSheet {
        let financialConnectionsSheet = FinancialConnectionsSheet(...)
        financialConnectionsSheet.apiClient = componentManager.apiClient.makeCopy()
        financialConnectionsSheet.apiClient.stripeAccount = ...
        financialConnectionsSheet.apiClient.publishableKey = ...
        return financialConnectionsSheet
    }

    func presentForToken(...) {
        let financialConnectionsSheet = makeSheet()
        ...
    }
}

class FinancialConnectionsPresenterTests: XCTestCase {
    func testMakeSheet() {
        let sheet = FinancialConnectionsPresenter().makeSheet(...)
        // Verify APIClient configured on `sheet`
    }
}
```

Reply:
This is definitely simpler, but it does have the drawback of not being able to test that the client secret was set correctly.

Reply:
Why not? The `financialConnectionsSessionClientSecret` property is publicly accessible, so we should able to XCAssertEqual on it.
https://github.com/stripe/stripe-ios/blob/d623be31/StripeFinancialConnections/StripeFinancialConnections/Source/FinancialConnectionsSheet.swift#L53

Reply:
🤦‍♂️ Oops, I was looking for clientSecret.